main.cpp

```diff
@@ -1201,46 +1210,68 @@ void manual_start_program(byte pid, byte uwt) {
 // ====== PUSH NOTIFICATION FUNCTIONS =======
 // ==========================================
 void ip2string(char* str, byte ip[4]) {
-  for(byte i=0;i<4;i++) {
-    itoa(ip[i], str+strlen(str), 10);
-    if(i!=3) strcat(str, ".");
-  }
+  sprintf_P(str+strlen(str), PSTR("%d.%d.%d.%d"), ip[0], ip[1], ip[2], ip[3]);
```

Comment:
I agree. I did initially this change because my compiler was complaining about itoa and remembered that sprintf is better approach in C. Then I though it was maybe a bad idea for arduino/esp OS. Impact must definitely be tested.

Reply:
sprintf is fine now: from firmware 2.1.9 we no longer supports legacy OS (any version prior to OS 2.3). Any current OS hardware has plenty of RAM and program space, so we are no longer limited by memory.